### DISCUSSION TITLE
Doesn't record system sound, sometimes not even mic (macOS 11.0.1)

### DISCUSSION THREAD

**Discussion by dialex** [ExistentialAudio/BlackHole#244] [General]
_Disclaimer: I'm terribly frustrated. I've followed dozens of tutorials and videos and I can't figure out how to make this work. After a couple of hours struggling, [this was the latest guide](https://wikis.utexas.edu/display/comm/How+to+set+up+BlackHole+Audio+on+a+Mac)._

I'm using macOS 11.0.1 Big Sur. I had this briefly working but now it just doesn't work:

- When I select the Multi-output device, I stop hearing sound
  - _even though I'm using the multi-aggregate device_
- My recordings don't record the system sound
- When I connect my headset, it doesn't record system sounds nor microphone

I've given up on my headset. I've tried Blackhole 2ch and 16ch. I just want this to work with the built-in mac...

<img width="880" alt="Screenshot 2020-12-14 at 18 03 02" src="https://user-images.githubusercontent.com/2140450/102118027-0a306800-3e37-11eb-8cb0-4421eb84cb0e.png">

Notice the `Out 0` on Macbook's Microphone. Is that normal? I can't change it.

<img width="880" alt="Screenshot 2020-12-14 at 18 03 04" src="https://user-images.githubusercontent.com/2140450/102118038-0dc3ef00-3e37-11eb-9fe1-27e3a0586512.png">

And when recording

<img width="780" alt="Screenshot 2020-12-14 at 18 10 53" src="https://user-images.githubusercontent.com/2140450/102118542-c12ce380-3e37-11eb-92c4-04c741e14c27.png">

<img width="780" alt="Screenshot 2020-12-14 at 18 10 55" src="https://user-images.githubusercontent.com/2140450/102118552-c427d400-3e37-11eb-90f1-d21453b73fdf.png">

Can someone please help me figure out what I'm doing wrong? I know this is open-source and free, and I have no right to complain. What frustrates me is that everyone else seems to have it working except me, and I had it working a single time and never again. For it to be so hard it must be something really simple.

**Comment by dialex** [comment]
Sorry, I just can't try more combinations on my own, I'm losing my mind. Reset.

1. I want to record the system sounds and hear them at the same time.
1. If possible record my mic.
1. If possible do all this while wearing a headset.

Please, how can I do that? 😭

**Comment by dialex** [comment]
Ok I managed to achieve 1)

<img width="880" alt="Screenshot 2020-12-14 at 18 59 34" src="https://user-images.githubusercontent.com/2140450/102123364-b295fa80-3e3e-11eb-8d79-375a2f4e64a0.png">

<img width="880" alt="Screenshot 2020-12-14 at 18 59 36" src="https://user-images.githubusercontent.com/2140450/102123413-b590eb00-3e3e-11eb-8fe2-8cf7ad5de549.png">

And when recording:
- Preferences > Sound > Multi-output device
- Quicktime player > Input > Aggregate Device

**Comment by devinroth** [comment]
Download BlackHole 2ch version. QT likes working with that one better. As well as multi-outputs. That may be part of your issues.

**Comment by devinroth** [comment]
Input into QT should be BH.

**Comment by devinroth** [comment]
System output should be the multi-output with BH and speakers/headphones.

**Comment by dialex** [comment]
To achieve the same, but using headphones, change the config to:

<img width="487" alt="Screenshot 2020-12-14 at 19 04 50" src="https://user-images.githubusercontent.com/2140450/102123776-436cd600-3e3f-11eb-8228-f9286168e439.png">

BUT!
- you need change the config while the Multi-Output Device (MOD) is not being used
- you need to keep the order Mac > BH > Headphones

Otherwise you mess up your MOD forever, and you need to delete it and start all over.

**Comment by devinroth** [comment]
The reason is due to a bug in macOS. That won't be a problem with the 2ch version.

**Comment by devinroth** [comment]
For recording your voice, you need to mix your mic into BH. I use Logic, but it should work with Audacity, Garage, or any other type of DAW. Just set the input from your mic, output to BH and enable monitoring so the audio gets passed through.
